covid_moonshot/analysis/plots.py

```diff
@@ -310,6 +311,88 @@ def plot_cumulative_distribution(
     plt.ylabel("Cumulative $N$ ligands")
 
 
+def plot_bootstrapped_clones(
+    complex_phase: PhaseAnalysis,
+    solvent_phase: PhaseAnalysis,
+    clones_per_gen: int,
+    n_gens: range,
+    n_bootstrap: int = 100,
+):
+    """
+    Plot free energy convergence with number of CLONEs
+
+    Parameters
+    ----------
+    complex_phase : PhaseAnalysis
+        results for complex
+    solvent_phase : PhaseAnalysis
+        results for solvent
+    clones_per_gen : int
+        Number of CLONEs per GEN
+    n_gens : range
+        Range of GENs for a RUN
+    n_bootstrap : int
+        Number of bootstrap samples
+
+    """
+
+    fig, ax = plt.subplots()
+
+    for n in n_gens:
+
+        # bootstrap
+        fes = []
+
+        for _ in range(n_bootstrap):
+
+            random_indices = np.random.choice(clones_per_gen, n)
+
+            subset_f = [
+                works.forward_works[x]
+                for x in random_indices
+                for works in complex_phase.gens
+            ]
+            subset_r = [
+                works.reverse_works[x]
+                for x in random_indices
+                for works in complex_phase.gens
+            ]
+
+            fe, _ = BAR(np.asarray(subset_f), np.asarray(subset_r))
+            fes.append(fe * KT_KCALMOL)
+
+        plt.scatter(n, np.mean(fes), color="red", label="complex")
+        plt.errorbar(n, np.mean(fes), yerr=np.std(fes), c="red")
+
+        fes = []
+
+        for _ in range(n_bootstrap):
+
+            random_indices = np.random.choice(clones_per_gen, n)
+            subset_f = [
+                works.forward_works[x]
+                for x in random_indices
+                for works in solvent_phase.gens
+            ]
+            subset_r = [
+                works.reverse_works[x]
+                for x in random_indices
+                for works in solvent_phase.gens
+            ]
+            fe, _ = BAR(np.asarray(subset_f), np.asarray(subset_r))
+            fes.append(fe * KT_KCALMOL)
```

Comment:
DRY suggestion: move this code into a function with a signature like
```python
def bootstrap(free_energies: List[FreeEnergy]) -> List[float]
```
and then reuse for complex and solvent
```
fe_complex = bootstrap(complex_phase.gens)
fe_solvent = bootstrap(solvent_phase.gens)
```

Reply:
Yeah good idea, would this be best placed in `free_energy.py` or kepy within `plots.py`?

Reply:
@mcwitt this has now been implemented!